plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/webview/LoginBrowser.kt

```diff
@@ -401,4 +412,13 @@ abstract class LoginBrowser(
             """.trimIndent()
         }
     }
+
+    fun isPKCEorDAG(region: String = "us-east-1"): AuthType {
+        val isCommercialRegion = !region.startsWith("us-gov") && !region.startsWith("us-iso") && !region.startsWith("cn")
+        if (!Registry.`is`("aws.dev.useDAG") && isCommercialRegion) {
+            return AuthType.PKCE
+        } else {
+            return AuthType.DeviceCode
+        }
+    }
```

Comment:
make `SsoAccessTokenProvider` use this logic too

Reply:
SsoAccessTokenProvider is not using this metric. SsoLoginCallBackProvider implements the copyCode functionality which is restricted to device code hence didnt add this login there

Reply:
sorry i meant just do that we have 1 util doing the PKCE or DAG decision and not needing to remember to do it in two places

Reply:
Oh moved the function to a util file